drivers/serial/uart_nrfx_uart.c

```diff
@@ -199,6 +258,8 @@ static int uart_nrfx_fifo_read(struct device *dev,
 {
 	u8_t num_rx = 0;
 
+	device_busy_set(dev);
```

Comment:
I see no point in calling it here. For this to work properly, the busy status would need to be cleared before this function returns (now it is cleared at the exit from the interrupt callback, but as I explained in the previous comment this needs to be changed). But is there a need to mark this device as busy only for the time while RX interrupt is being served?